wfc_core/src/bitvec.rs

```diff
@@ -153,41 +175,80 @@ impl<'a> Iterator for BitVecIterator<'a> {
         // over, we mask them. Masks are computed from the the index to the bit
         // we would visit next.
 
-        let n = u32::from(self.next);
+        // TODO(yan): @Correctness self.next likely won't be larger than than
+        // u32, but we'd love to prove this statically. Unfortunately, we need
+        // to bound N with usize, because it's a parameter for array size.
```

Comment:
We can ensure this in caller

Reply:
This is already runtime-ensured in upper layers of the library here too. This is just my OCD that I'd like it to be provable by types and having less casts overall.